api/src/lib/github/repository.go

```diff
@@ -0,0 +1,59 @@
+package github
+
+import (
+	"strconv"
+	"io/ioutil"
+	"encoding/json"
+	"net/http"
+	"porter/api/src/config"
+	)
+
+type RepositoryResponse struct {
+	Url string `json:"url"`
+	Name string `json:"name"`
+	Html_url string `json:"html_url"`
+}
+
+type Repository struct {
+	Name         string
+	ExposedPorts []string
+	Url 		string
+}
+
+
+func (rp Repository) MarshalBinary() ([]byte, error) {
+	return json.Marshal(&rp)
+}
+
+func (rp *Repository) UnMarshalBinary(data []byte) (error) {
+	return json.Unmarshal(data, &rp)
+}
+
+func FetchRepository(page int,ch chan []RepositoryResponse){
+	var repositoryResponse []RepositoryResponse
+	resp,_ := http.Get("https://api.github.com/orgs/fiverr/repos?access_token="+ config.GetValue("githubToken") +"&limit=100&sort=pushed&page=" + strconv.Itoa(page))
+	body,_ := ioutil.ReadAll(resp.Body)
+	json.Unmarshal(body, &repositoryResponse)
+	ch <- repositoryResponse
+}
+
+func FetchAllRepositories() ([]RepositoryResponse) {
+	repositoriesResponseCh := make(chan []RepositoryResponse)
+	for i:= 1; i <= 20 ; i++  {
```

Comment:
I should fetch the repositoryCount and divide it for each page count, for now it's ok